==22249== Memcheck, a memory error detector
==22249== Copyright (C) 2002-2017, and GNU GPL'd, by Julian Seward et al.
==22249== Using Valgrind-3.13.0 and LibVEX; rerun with -h for copyright info
==22249== Command: /home/wasmer/repositories/wasmer/wasmer/target/release/examples/early-exit
==22249== 
==22249== Conditional jump or move depends on uninitialised value(s)
==22249==    at 0x68F06D: wasmparser::validator::Validator::end (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/early-exit)
==22249==    by 0x684B6C: wasmparser::validator::Validator::payload (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/early-exit)
==22249==    by 0x683DE1: wasmparser::validator::Validator::validate_all (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/early-exit)
==22249==    by 0x29D57C: wasmer_compiler::compiler::Compiler::validate_module (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/early-exit)
==22249==    by 0x268E68: <wasmer_engine_jit::engine::JITEngine as wasmer_engine::engine::Engine>::validate (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/early-exit)
==22249==    by 0x57F9E4: wasmer::module::Module::from_binary (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/early-exit)
==22249==    by 0x260201: wasmer::module::Module::new (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/early-exit)
==22249==    by 0x263B46: early_exit::main (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/early-exit)
==22249==    by 0x262A66: std::rt::lang_start::{{closure}} (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/early-exit)
==22249==    by 0x732272: {{closure}} (rt.rs:52)
==22249==    by 0x732272: do_call<closure-0,i32> (panicking.rs:348)
==22249==    by 0x732272: try<i32,closure-0> (panicking.rs:325)
==22249==    by 0x732272: catch_unwind<closure-0,i32> (panic.rs:394)
==22249==    by 0x732272: std::rt::lang_start_internal (rt.rs:51)
==22249==    by 0x264881: main (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/early-exit)
==22249==  Uninitialised value was created by a stack allocation
==22249==    at 0x29D490: wasmer_compiler::compiler::Compiler::validate_module (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/early-exit)
==22249== 
==22249== 
==22249== HEAP SUMMARY:
==22249==     in use at exit: 30,384 bytes in 91 blocks
==22249==   total heap usage: 763 allocs, 672 frees, 147,111 bytes allocated
==22249== 
==22249== 16 bytes in 1 blocks are still reachable in loss record 1 of 31
==22249==    at 0x4C31B0F: malloc (in /usr/lib/valgrind/vgpreload_memcheck-amd64-linux.so)
==22249==    by 0x6C2FE8: std::sync::once::Once::call_once::{{closure}} (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/early-exit)
==22249==    by 0x72B919: std::sync::once::Once::call_inner (once.rs:416)
==22249==    by 0x6C7826: backtrace::lock::lock (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/early-exit)
==22249==    by 0x6CF970: backtrace::backtrace::trace (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/early-exit)
==22249==    by 0x6C2535: backtrace::capture::Backtrace::new_unresolved (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/early-exit)
==22249==    by 0x647B0A: wasmer_engine::trap::error::RuntimeError::from_trap (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/early-exit)
==22249==    by 0x648A15: <wasmer_engine::trap::error::RuntimeError as core::convert::From<wasmer_vm::trap::traphandlers::Trap>>::from (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/early-exit)
==22249==    by 0x265000: wasmer::native::NativeFunc<(A1,A2),Rets>::call (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/early-exit)
==22249==    by 0x264101: early_exit::main (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/early-exit)
==22249==    by 0x262A66: std::rt::lang_start::{{closure}} (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/early-exit)
==22249==    by 0x732272: {{closure}} (rt.rs:52)
==22249==    by 0x732272: do_call<closure-0,i32> (panicking.rs:348)
==22249==    by 0x732272: try<i32,closure-0> (panicking.rs:325)
==22249==    by 0x732272: catch_unwind<closure-0,i32> (panic.rs:394)
==22249==    by 0x732272: std::rt::lang_start_internal (rt.rs:51)
==22249== 
==22249== 32 bytes in 1 blocks are still reachable in loss record 2 of 31
==22249==    at 0x4C33B25: calloc (in /usr/lib/valgrind/vgpreload_memcheck-amd64-linux.so)
==22249==    by 0x4E3F7F4: _dlerror_run (dlerror.c:140)
==22249==    by 0x4E3F165: dlsym (dlsym.c:70)
==22249==    by 0x740843: fetch (weak.rs:61)
==22249==    by 0x740843: get<unsafe extern "C" fn(*const libc::unix::linux_like::linux::gnu::b64::x86_64::pthread_attr_t) -> usize> (weak.rs:46)
==22249==    by 0x740843: min_stack_size (thread.rs:449)
==22249==    by 0x740843: std::sys::unix::thread::Thread::new (thread.rs:49)
==22249==    by 0x2E0D76: std::thread::Builder::spawn (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/early-exit)
==22249==    by 0x2E4016: <rayon_core::registry::DefaultSpawn as rayon_core::registry::ThreadSpawn>::spawn (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/early-exit)
==22249==    by 0x2E4BDF: rayon_core::registry::Registry::new (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/early-exit)
==22249==    by 0x2E7D65: std::sync::once::Once::call_once::{{closure}} (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/early-exit)
==22249==    by 0x72B919: std::sync::once::Once::call_inner (once.rs:416)
==22249==    by 0x2E4195: rayon_core::registry::global_registry (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/early-exit)
==22249==    by 0x2E7144: rayon_core::current_num_threads (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/early-exit)
==22249==    by 0x290541: <rayon::iter::plumbing::bridge::Callback<C> as rayon::iter::plumbing::ProducerCallback<I>>::callback (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/early-exit)
==22249== 
==22249== 40 bytes in 1 blocks are still reachable in loss record 3 of 31
==22249==    at 0x4C31B0F: malloc (in /usr/lib/valgrind/vgpreload_memcheck-amd64-linux.so)
==22249==    by 0x6CEFCB: std::sync::mutex::Mutex<T>::new (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/early-exit)
==22249==    by 0x6C2FD8: std::sync::once::Once::call_once::{{closure}} (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/early-exit)
==22249==    by 0x72B919: std::sync::once::Once::call_inner (once.rs:416)
==22249==    by 0x6C7826: backtrace::lock::lock (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/early-exit)
==22249==    by 0x6CF970: backtrace::backtrace::trace (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/early-exit)
==22249==    by 0x6C2535: backtrace::capture::Backtrace::new_unresolved (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/early-exit)
==22249==    by 0x647B0A: wasmer_engine::trap::error::RuntimeError::from_trap (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/early-exit)
==22249==    by 0x648A15: <wasmer_engine::trap::error::RuntimeError as core::convert::From<wasmer_vm::trap::traphandlers::Trap>>::from (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/early-exit)
==22249==    by 0x265000: wasmer::native::NativeFunc<(A1,A2),Rets>::call (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/early-exit)
==22249==    by 0x264101: early_exit::main (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/early-exit)
==22249==    by 0x262A66: std::rt::lang_start::{{closure}} (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/early-exit)
==22249== 
==22249== 64 bytes in 4 blocks are still reachable in loss record 4 of 31
==22249==    at 0x4C31B0F: malloc (in /usr/lib/valgrind/vgpreload_memcheck-amd64-linux.so)
==22249==    by 0x2EA02C: crossbeam_deque::deque::Worker<T>::new_lifo (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/early-exit)
==22249==    by 0x2EB017: core::iter::traits::iterator::Iterator::unzip (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/early-exit)
==22249==    by 0x2E44B6: rayon_core::registry::Registry::new (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/early-exit)
==22249==    by 0x2E7D65: std::sync::once::Once::call_once::{{closure}} (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/early-exit)
==22249==    by 0x72B919: std::sync::once::Once::call_inner (once.rs:416)
==22249==    by 0x2E4195: rayon_core::registry::global_registry (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/early-exit)
==22249==    by 0x2E7144: rayon_core::current_num_threads (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/early-exit)
==22249==    by 0x290541: <rayon::iter::plumbing::bridge::Callback<C> as rayon::iter::plumbing::ProducerCallback<I>>::callback (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/early-exit)
==22249==    by 0x296AB5: <rayon::iter::while_some::WhileSome<I> as rayon::iter::ParallelIterator>::drive_unindexed (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/early-exit)
==22249==    by 0x29AFB8: rayon::iter::collect::<impl rayon::iter::ParallelExtend<T> for alloc::vec::Vec<T>>::par_extend (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/early-exit)
==22249==    by 0x2B6FF1: rayon::result::<impl rayon::iter::FromParallelIterator<core::result::Result<T,E>> for core::result::Result<C,E>>::from_par_iter (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/early-exit)
==22249== 
==22249== 64 bytes in 4 blocks are still reachable in loss record 5 of 31
==22249==    at 0x4C31B0F: malloc (in /usr/lib/valgrind/vgpreload_memcheck-amd64-linux.so)
==22249==    by 0x74079C: alloc (alloc.rs:80)
==22249==    by 0x74079C: alloc (alloc.rs:174)
==22249==    by 0x74079C: exchange_malloc (alloc.rs:277)
==22249==    by 0x74079C: std::sys::unix::thread::Thread::new (thread.rs:44)
==22249==    by 0x2E0D76: std::thread::Builder::spawn (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/early-exit)
==22249==    by 0x2E4016: <rayon_core::registry::DefaultSpawn as rayon_core::registry::ThreadSpawn>::spawn (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/early-exit)
==22249==    by 0x2E4BDF: rayon_core::registry::Registry::new (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/early-exit)
==22249==    by 0x2E7D65: std::sync::once::Once::call_once::{{closure}} (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/early-exit)
==22249==    by 0x72B919: std::sync::once::Once::call_inner (once.rs:416)
==22249==    by 0x2E4195: rayon_core::registry::global_registry (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/early-exit)
==22249==    by 0x2E7144: rayon_core::current_num_threads (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/early-exit)
==22249==    by 0x290541: <rayon::iter::plumbing::bridge::Callback<C> as rayon::iter::plumbing::ProducerCallback<I>>::callback (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/early-exit)
==22249==    by 0x296AB5: <rayon::iter::while_some::WhileSome<I> as rayon::iter::ParallelIterator>::drive_unindexed (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/early-exit)
==22249==    by 0x29AFB8: rayon::iter::collect::<impl rayon::iter::ParallelExtend<T> for alloc::vec::Vec<T>>::par_extend (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/early-exit)
==22249== 
==22249== 72 bytes in 1 blocks are still reachable in loss record 6 of 31
==22249==    at 0x4C31B0F: malloc (in /usr/lib/valgrind/vgpreload_memcheck-amd64-linux.so)
==22249==    by 0x6468AD: std::sync::once::Once::call_once::{{closure}} (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/early-exit)
==22249==    by 0x72B919: std::sync::once::Once::call_inner (once.rs:416)
==22249==    by 0x64A51B: wasmer_engine::trap::frame_info::register (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/early-exit)
==22249==    by 0x28A355: <wasmer_engine_jit::artifact::JITArtifact as wasmer_engine::artifact::Artifact>::register_frame_info (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/early-exit)
==22249==    by 0x284625: wasmer_engine::artifact::Artifact::instantiate (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/early-exit)
==22249==    by 0x57FD9E: wasmer::module::Module::instantiate (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/early-exit)
==22249==    by 0x57E713: wasmer::instance::Instance::new (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/early-exit)
==22249==    by 0x263D35: early_exit::main (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/early-exit)
==22249==    by 0x262A66: std::rt::lang_start::{{closure}} (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/early-exit)
==22249==    by 0x732272: {{closure}} (rt.rs:52)
==22249==    by 0x732272: do_call<closure-0,i32> (panicking.rs:348)
==22249==    by 0x732272: try<i32,closure-0> (panicking.rs:325)
==22249==    by 0x732272: catch_unwind<closure-0,i32> (panic.rs:394)
==22249==    by 0x732272: std::rt::lang_start_internal (rt.rs:51)
==22249==    by 0x264881: main (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/early-exit)
==22249== 
==22249== 128 bytes in 4 blocks are still reachable in loss record 7 of 31
==22249==    at 0x4C33B25: calloc (in /usr/lib/valgrind/vgpreload_memcheck-amd64-linux.so)
==22249==    by 0x56C48E1: __cxa_thread_atexit_impl (cxa_thread_atexit_impl.c:106)
==22249==    by 0x70CDCE: register_dtor (fast_thread_local.rs:33)
==22249==    by 0x70CDCE: try_register_dtor<core::cell::RefCell<core::option::Option<std::sys_common::thread_info::ThreadInfo>>> (local.rs:438)
==22249==    by 0x70CDCE: std::thread::local::fast::Key<T>::try_initialize (local.rs:424)
==22249==    by 0x72F5C9: get<core::cell::RefCell<core::option::Option<std::sys_common::thread_info::ThreadInfo>>,fn() -> core::cell::RefCell<core::option::Option<std::sys_common::thread_info::ThreadInfo>>> (local.rs:410)
==22249==    by 0x72F5C9: __getit (local.rs:175)
==22249==    by 0x72F5C9: try_with<core::cell::RefCell<core::option::Option<std::sys_common::thread_info::ThreadInfo>>,closure-0,()> (local.rs:262)
==22249==    by 0x72F5C9: with<core::cell::RefCell<core::option::Option<std::sys_common::thread_info::ThreadInfo>>,closure-0,()> (local.rs:239)
==22249==    by 0x72F5C9: std::sys_common::thread_info::set (thread_info.rs:40)
==22249==    by 0x2E0EFC: core::ops::function::FnOnce::call_once{{vtable-shim}} (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/early-exit)
==22249==    by 0x740BB9: call_once<(),FnOnce<()>> (boxed.rs:1081)
==22249==    by 0x740BB9: call_once<(),alloc::boxed::Box<FnOnce<()>>> (boxed.rs:1081)
==22249==    by 0x740BB9: std::sys::unix::thread::Thread::new::thread_start (thread.rs:87)
==22249==    by 0x52516DA: start_thread (pthread_create.c:463)
==22249==    by 0x57A271E: clone (clone.S:95)
==22249== 
==22249== 128 bytes in 4 blocks are still reachable in loss record 8 of 31
==22249==    at 0x4C33B25: calloc (in /usr/lib/valgrind/vgpreload_memcheck-amd64-linux.so)
==22249==    by 0x56C48E1: __cxa_thread_atexit_impl (cxa_thread_atexit_impl.c:106)
==22249==    by 0x2F27ED: _ZN3std6thread5local4fast12Key$LT$T$GT$14try_initialize17h3d6d30c5335b0a67E.llvm.12191645446024711930 (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/early-exit)
==22249==    by 0x2EA1D2: crossbeam_deque::deque::Stealer<T>::steal (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/early-exit)
==22249==    by 0x2E1C96: <core::iter::adapters::chain::Chain<A,B> as core::iter::traits::iterator::Iterator>::try_fold (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/early-exit)
==22249==    by 0x2E5929: rayon_core::registry::WorkerThread::wait_until_cold (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/early-exit)
==22249==    by 0x2E39FA: rayon_core::registry::ThreadBuilder::run (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/early-exit)
==22249==    by 0x2E6640: std::sys_common::backtrace::__rust_begin_short_backtrace (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/early-exit)
==22249==    by 0x2E0F3E: core::ops::function::FnOnce::call_once{{vtable-shim}} (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/early-exit)
==22249==    by 0x740BB9: call_once<(),FnOnce<()>> (boxed.rs:1081)
==22249==    by 0x740BB9: call_once<(),alloc::boxed::Box<FnOnce<()>>> (boxed.rs:1081)
==22249==    by 0x740BB9: std::sys::unix::thread::Thread::new::thread_start (thread.rs:87)
==22249==    by 0x52516DA: start_thread (pthread_create.c:463)
==22249==    by 0x57A271E: clone (clone.S:95)
==22249== 
==22249== 160 bytes in 4 blocks are still reachable in loss record 9 of 31
==22249==    at 0x4C31B0F: malloc (in /usr/lib/valgrind/vgpreload_memcheck-amd64-linux.so)
==22249==    by 0x2E815C: std::sync::mutex::Mutex<T>::new (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/early-exit)
==22249==    by 0x2E552B: rayon_core::registry::ThreadInfo::new (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/early-exit)
==22249==    by 0x2EB398: <core::iter::adapters::Map<I,F> as core::iter::traits::iterator::Iterator>::fold (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/early-exit)
==22249==    by 0x2E6460: <alloc::vec::Vec<T> as alloc::vec::SpecExtend<T,I>>::from_iter (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/early-exit)
==22249==    by 0x2E455A: rayon_core::registry::Registry::new (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/early-exit)
==22249==    by 0x2E7D65: std::sync::once::Once::call_once::{{closure}} (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/early-exit)
==22249==    by 0x72B919: std::sync::once::Once::call_inner (once.rs:416)
==22249==    by 0x2E4195: rayon_core::registry::global_registry (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/early-exit)
==22249==    by 0x2E7144: rayon_core::current_num_threads (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/early-exit)
==22249==    by 0x290541: <rayon::iter::plumbing::bridge::Callback<C> as rayon::iter::plumbing::ProducerCallback<I>>::callback (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/early-exit)
==22249==    by 0x296AB5: <rayon::iter::while_some::WhileSome<I> as rayon::iter::ParallelIterator>::drive_unindexed (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/early-exit)
==22249== 
==22249== 160 bytes in 4 blocks are still reachable in loss record 10 of 31
==22249==    at 0x4C31B0F: malloc (in /usr/lib/valgrind/vgpreload_memcheck-amd64-linux.so)
==22249==    by 0x2E815C: std::sync::mutex::Mutex<T>::new (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/early-exit)
==22249==    by 0x2E5556: rayon_core::registry::ThreadInfo::new (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/early-exit)
==22249==    by 0x2EB398: <core::iter::adapters::Map<I,F> as core::iter::traits::iterator::Iterator>::fold (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/early-exit)
==22249==    by 0x2E6460: <alloc::vec::Vec<T> as alloc::vec::SpecExtend<T,I>>::from_iter (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/early-exit)
==22249==    by 0x2E455A: rayon_core::registry::Registry::new (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/early-exit)
==22249==    by 0x2E7D65: std::sync::once::Once::call_once::{{closure}} (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/early-exit)
==22249==    by 0x72B919: std::sync::once::Once::call_inner (once.rs:416)
==22249==    by 0x2E4195: rayon_core::registry::global_registry (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/early-exit)
==22249==    by 0x2E7144: rayon_core::current_num_threads (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/early-exit)
==22249==    by 0x290541: <rayon::iter::plumbing::bridge::Callback<C> as rayon::iter::plumbing::ProducerCallback<I>>::callback (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/early-exit)
==22249==    by 0x296AB5: <rayon::iter::while_some::WhileSome<I> as rayon::iter::ParallelIterator>::drive_unindexed (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/early-exit)
==22249== 
==22249== 160 bytes in 4 blocks are still reachable in loss record 11 of 31
==22249==    at 0x4C31B0F: malloc (in /usr/lib/valgrind/vgpreload_memcheck-amd64-linux.so)
==22249==    by 0x2E815C: std::sync::mutex::Mutex<T>::new (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/early-exit)
==22249==    by 0x2EB1FB: <core::iter::adapters::Map<I,F> as core::iter::traits::iterator::Iterator>::fold (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/early-exit)
==22249==    by 0x2E45D0: rayon_core::registry::Registry::new (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/early-exit)
==22249==    by 0x2E7D65: std::sync::once::Once::call_once::{{closure}} (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/early-exit)
==22249==    by 0x72B919: std::sync::once::Once::call_inner (once.rs:416)
==22249==    by 0x2E4195: rayon_core::registry::global_registry (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/early-exit)
==22249==    by 0x2E7144: rayon_core::current_num_threads (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/early-exit)
==22249==    by 0x290541: <rayon::iter::plumbing::bridge::Callback<C> as rayon::iter::plumbing::ProducerCallback<I>>::callback (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/early-exit)
==22249==    by 0x296AB5: <rayon::iter::while_some::WhileSome<I> as rayon::iter::ParallelIterator>::drive_unindexed (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/early-exit)
==22249==    by 0x29AFB8: rayon::iter::collect::<impl rayon::iter::ParallelExtend<T> for alloc::vec::Vec<T>>::par_extend (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/early-exit)
==22249==    by 0x2B6FF1: rayon::result::<impl rayon::iter::FromParallelIterator<core::result::Result<T,E>> for core::result::Result<C,E>>::from_par_iter (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/early-exit)
==22249== 
==22249== 160 bytes in 4 blocks are still reachable in loss record 12 of 31
==22249==    at 0x4C31B0F: malloc (in /usr/lib/valgrind/vgpreload_memcheck-amd64-linux.so)
==22249==    by 0x70E16E: alloc (alloc.rs:80)
==22249==    by 0x70E16E: alloc (alloc.rs:174)
==22249==    by 0x70E16E: exchange_malloc (alloc.rs:277)
==22249==    by 0x70E16E: new<()> (mutex.rs:223)
==22249==    by 0x70E16E: std::thread::Thread::new (mod.rs:1149)
==22249==    by 0x2E0C61: std::thread::Builder::spawn (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/early-exit)
==22249==    by 0x2E4016: <rayon_core::registry::DefaultSpawn as rayon_core::registry::ThreadSpawn>::spawn (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/early-exit)
==22249==    by 0x2E4BDF: rayon_core::registry::Registry::new (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/early-exit)
==22249==    by 0x2E7D65: std::sync::once::Once::call_once::{{closure}} (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/early-exit)
==22249==    by 0x72B919: std::sync::once::Once::call_inner (once.rs:416)
==22249==    by 0x2E4195: rayon_core::registry::global_registry (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/early-exit)
==22249==    by 0x2E7144: rayon_core::current_num_threads (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/early-exit)
==22249==    by 0x290541: <rayon::iter::plumbing::bridge::Callback<C> as rayon::iter::plumbing::ProducerCallback<I>>::callback (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/early-exit)
==22249==    by 0x296AB5: <rayon::iter::while_some::WhileSome<I> as rayon::iter::ParallelIterator>::drive_unindexed (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/early-exit)
==22249==    by 0x29AFB8: rayon::iter::collect::<impl rayon::iter::ParallelExtend<T> for alloc::vec::Vec<T>>::par_extend (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/early-exit)
==22249== 
==22249== 160 bytes in 4 blocks are still reachable in loss record 13 of 31
==22249==    at 0x4C31B0F: malloc (in /usr/lib/valgrind/vgpreload_memcheck-amd64-linux.so)
==22249==    by 0x2E0C83: std::thread::Builder::spawn (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/early-exit)
==22249==    by 0x2E4016: <rayon_core::registry::DefaultSpawn as rayon_core::registry::ThreadSpawn>::spawn (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/early-exit)
==22249==    by 0x2E4BDF: rayon_core::registry::Registry::new (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/early-exit)
==22249==    by 0x2E7D65: std::sync::once::Once::call_once::{{closure}} (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/early-exit)
==22249==    by 0x72B919: std::sync::once::Once::call_inner (once.rs:416)
==22249==    by 0x2E4195: rayon_core::registry::global_registry (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/early-exit)
==22249==    by 0x2E7144: rayon_core::current_num_threads (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/early-exit)
==22249==    by 0x290541: <rayon::iter::plumbing::bridge::Callback<C> as rayon::iter::plumbing::ProducerCallback<I>>::callback (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/early-exit)
==22249==    by 0x296AB5: <rayon::iter::while_some::WhileSome<I> as rayon::iter::ParallelIterator>::drive_unindexed (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/early-exit)
==22249==    by 0x29AFB8: rayon::iter::collect::<impl rayon::iter::ParallelExtend<T> for alloc::vec::Vec<T>>::par_extend (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/early-exit)
==22249==    by 0x2B6FF1: rayon::result::<impl rayon::iter::FromParallelIterator<core::result::Result<T,E>> for core::result::Result<C,E>>::from_par_iter (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/early-exit)
==22249== 
==22249== 192 bytes in 4 blocks are still reachable in loss record 14 of 31
==22249==    at 0x4C31B0F: malloc (in /usr/lib/valgrind/vgpreload_memcheck-amd64-linux.so)
==22249==    by 0x72B266: alloc (alloc.rs:80)
==22249==    by 0x72B266: alloc (alloc.rs:174)
==22249==    by 0x72B266: exchange_malloc (alloc.rs:277)
==22249==    by 0x72B266: std::sync::condvar::Condvar::new (condvar.rs:130)
==22249==    by 0x2E5536: rayon_core::registry::ThreadInfo::new (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/early-exit)
==22249==    by 0x2EB398: <core::iter::adapters::Map<I,F> as core::iter::traits::iterator::Iterator>::fold (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/early-exit)
==22249==    by 0x2E6460: <alloc::vec::Vec<T> as alloc::vec::SpecExtend<T,I>>::from_iter (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/early-exit)
==22249==    by 0x2E455A: rayon_core::registry::Registry::new (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/early-exit)
==22249==    by 0x2E7D65: std::sync::once::Once::call_once::{{closure}} (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/early-exit)
==22249==    by 0x72B919: std::sync::once::Once::call_inner (once.rs:416)
==22249==    by 0x2E4195: rayon_core::registry::global_registry (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/early-exit)
==22249==    by 0x2E7144: rayon_core::current_num_threads (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/early-exit)
==22249==    by 0x290541: <rayon::iter::plumbing::bridge::Callback<C> as rayon::iter::plumbing::ProducerCallback<I>>::callback (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/early-exit)
==22249==    by 0x296AB5: <rayon::iter::while_some::WhileSome<I> as rayon::iter::ParallelIterator>::drive_unindexed (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/early-exit)
==22249== 
==22249== 192 bytes in 4 blocks are still reachable in loss record 15 of 31
==22249==    at 0x4C31B0F: malloc (in /usr/lib/valgrind/vgpreload_memcheck-amd64-linux.so)
==22249==    by 0x72B266: alloc (alloc.rs:80)
==22249==    by 0x72B266: alloc (alloc.rs:174)
==22249==    by 0x72B266: exchange_malloc (alloc.rs:277)
==22249==    by 0x72B266: std::sync::condvar::Condvar::new (condvar.rs:130)
==22249==    by 0x2E5561: rayon_core::registry::ThreadInfo::new (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/early-exit)
==22249==    by 0x2EB398: <core::iter::adapters::Map<I,F> as core::iter::traits::iterator::Iterator>::fold (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/early-exit)
==22249==    by 0x2E6460: <alloc::vec::Vec<T> as alloc::vec::SpecExtend<T,I>>::from_iter (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/early-exit)
==22249==    by 0x2E455A: rayon_core::registry::Registry::new (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/early-exit)
==22249==    by 0x2E7D65: std::sync::once::Once::call_once::{{closure}} (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/early-exit)
==22249==    by 0x72B919: std::sync::once::Once::call_inner (once.rs:416)
==22249==    by 0x2E4195: rayon_core::registry::global_registry (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/early-exit)
==22249==    by 0x2E7144: rayon_core::current_num_threads (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/early-exit)
==22249==    by 0x290541: <rayon::iter::plumbing::bridge::Callback<C> as rayon::iter::plumbing::ProducerCallback<I>>::callback (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/early-exit)
==22249==    by 0x296AB5: <rayon::iter::while_some::WhileSome<I> as rayon::iter::ParallelIterator>::drive_unindexed (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/early-exit)
==22249== 
==22249== 192 bytes in 4 blocks are still reachable in loss record 16 of 31
==22249==    at 0x4C31B0F: malloc (in /usr/lib/valgrind/vgpreload_memcheck-amd64-linux.so)
==22249==    by 0x72B266: alloc (alloc.rs:80)
==22249==    by 0x72B266: alloc (alloc.rs:174)
==22249==    by 0x72B266: exchange_malloc (alloc.rs:277)
==22249==    by 0x72B266: std::sync::condvar::Condvar::new (condvar.rs:130)
==22249==    by 0x2EB201: <core::iter::adapters::Map<I,F> as core::iter::traits::iterator::Iterator>::fold (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/early-exit)
==22249==    by 0x2E45D0: rayon_core::registry::Registry::new (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/early-exit)
==22249==    by 0x2E7D65: std::sync::once::Once::call_once::{{closure}} (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/early-exit)
==22249==    by 0x72B919: std::sync::once::Once::call_inner (once.rs:416)
==22249==    by 0x2E4195: rayon_core::registry::global_registry (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/early-exit)
==22249==    by 0x2E7144: rayon_core::current_num_threads (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/early-exit)
==22249==    by 0x290541: <rayon::iter::plumbing::bridge::Callback<C> as rayon::iter::plumbing::ProducerCallback<I>>::callback (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/early-exit)
==22249==    by 0x296AB5: <rayon::iter::while_some::WhileSome<I> as rayon::iter::ParallelIterator>::drive_unindexed (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/early-exit)
==22249==    by 0x29AFB8: rayon::iter::collect::<impl rayon::iter::ParallelExtend<T> for alloc::vec::Vec<T>>::par_extend (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/early-exit)
==22249==    by 0x2B6FF1: rayon::result::<impl rayon::iter::FromParallelIterator<core::result::Result<T,E>> for core::result::Result<C,E>>::from_par_iter (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/early-exit)
==22249== 
==22249== 192 bytes in 4 blocks are still reachable in loss record 17 of 31
==22249==    at 0x4C31B0F: malloc (in /usr/lib/valgrind/vgpreload_memcheck-amd64-linux.so)
==22249==    by 0x70E1D2: alloc (alloc.rs:80)
==22249==    by 0x70E1D2: alloc (alloc.rs:174)
==22249==    by 0x70E1D2: exchange_malloc (alloc.rs:277)
==22249==    by 0x70E1D2: new (condvar.rs:130)
==22249==    by 0x70E1D2: std::thread::Thread::new (mod.rs:1150)
==22249==    by 0x2E0C61: std::thread::Builder::spawn (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/early-exit)
==22249==    by 0x2E4016: <rayon_core::registry::DefaultSpawn as rayon_core::registry::ThreadSpawn>::spawn (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/early-exit)
==22249==    by 0x2E4BDF: rayon_core::registry::Registry::new (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/early-exit)
==22249==    by 0x2E7D65: std::sync::once::Once::call_once::{{closure}} (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/early-exit)
==22249==    by 0x72B919: std::sync::once::Once::call_inner (once.rs:416)
==22249==    by 0x2E4195: rayon_core::registry::global_registry (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/early-exit)
==22249==    by 0x2E7144: rayon_core::current_num_threads (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/early-exit)
==22249==    by 0x290541: <rayon::iter::plumbing::bridge::Callback<C> as rayon::iter::plumbing::ProducerCallback<I>>::callback (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/early-exit)
==22249==    by 0x296AB5: <rayon::iter::while_some::WhileSome<I> as rayon::iter::ParallelIterator>::drive_unindexed (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/early-exit)
==22249==    by 0x29AFB8: rayon::iter::collect::<impl rayon::iter::ParallelExtend<T> for alloc::vec::Vec<T>>::par_extend (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/early-exit)
==22249== 
==22249== 320 bytes in 4 blocks are still reachable in loss record 18 of 31
==22249==    at 0x4C31B0F: malloc (in /usr/lib/valgrind/vgpreload_memcheck-amd64-linux.so)
==22249==    by 0x70E221: alloc (alloc.rs:80)
==22249==    by 0x70E221: alloc (alloc.rs:174)
==22249==    by 0x70E221: exchange_malloc (alloc.rs:277)
==22249==    by 0x70E221: new<std::thread::Inner> (sync.rs:324)
==22249==    by 0x70E221: std::thread::Thread::new (mod.rs:1145)
==22249==    by 0x2E0C61: std::thread::Builder::spawn (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/early-exit)
==22249==    by 0x2E4016: <rayon_core::registry::DefaultSpawn as rayon_core::registry::ThreadSpawn>::spawn (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/early-exit)
==22249==    by 0x2E4BDF: rayon_core::registry::Registry::new (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/early-exit)
==22249==    by 0x2E7D65: std::sync::once::Once::call_once::{{closure}} (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/early-exit)
==22249==    by 0x72B919: std::sync::once::Once::call_inner (once.rs:416)
==22249==    by 0x2E4195: rayon_core::registry::global_registry (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/early-exit)
==22249==    by 0x2E7144: rayon_core::current_num_threads (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/early-exit)
==22249==    by 0x290541: <rayon::iter::plumbing::bridge::Callback<C> as rayon::iter::plumbing::ProducerCallback<I>>::callback (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/early-exit)
==22249==    by 0x296AB5: <rayon::iter::while_some::WhileSome<I> as rayon::iter::ParallelIterator>::drive_unindexed (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/early-exit)
==22249==    by 0x29AFB8: rayon::iter::collect::<impl rayon::iter::ParallelExtend<T> for alloc::vec::Vec<T>>::par_extend (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/early-exit)
==22249== 
==22249== 384 bytes in 1 blocks are still reachable in loss record 19 of 31
==22249==    at 0x4C31B0F: malloc (in /usr/lib/valgrind/vgpreload_memcheck-amd64-linux.so)
==22249==    by 0x2E967C: alloc::raw_vec::RawVec<T,A>::reserve (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/early-exit)
==22249==    by 0x2E641C: <alloc::vec::Vec<T> as alloc::vec::SpecExtend<T,I>>::from_iter (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/early-exit)
==22249==    by 0x2E455A: rayon_core::registry::Registry::new (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/early-exit)
==22249==    by 0x2E7D65: std::sync::once::Once::call_once::{{closure}} (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/early-exit)
==22249==    by 0x72B919: std::sync::once::Once::call_inner (once.rs:416)
==22249==    by 0x2E4195: rayon_core::registry::global_registry (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/early-exit)
==22249==    by 0x2E7144: rayon_core::current_num_threads (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/early-exit)
==22249==    by 0x290541: <rayon::iter::plumbing::bridge::Callback<C> as rayon::iter::plumbing::ProducerCallback<I>>::callback (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/early-exit)
==22249==    by 0x296AB5: <rayon::iter::while_some::WhileSome<I> as rayon::iter::ParallelIterator>::drive_unindexed (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/early-exit)
==22249==    by 0x29AFB8: rayon::iter::collect::<impl rayon::iter::ParallelExtend<T> for alloc::vec::Vec<T>>::par_extend (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/early-exit)
==22249==    by 0x2B6FF1: rayon::result::<impl rayon::iter::FromParallelIterator<core::result::Result<T,E>> for core::result::Result<C,E>>::from_par_iter (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/early-exit)
==22249== 
==22249== 416 bytes in 4 blocks are still reachable in loss record 20 of 31
==22249==    at 0x4C31B0F: malloc (in /usr/lib/valgrind/vgpreload_memcheck-amd64-linux.so)
==22249==    by 0x2E0D18: std::thread::Builder::spawn (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/early-exit)
==22249==    by 0x2E4016: <rayon_core::registry::DefaultSpawn as rayon_core::registry::ThreadSpawn>::spawn (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/early-exit)
==22249==    by 0x2E4BDF: rayon_core::registry::Registry::new (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/early-exit)
==22249==    by 0x2E7D65: std::sync::once::Once::call_once::{{closure}} (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/early-exit)
==22249==    by 0x72B919: std::sync::once::Once::call_inner (once.rs:416)
==22249==    by 0x2E4195: rayon_core::registry::global_registry (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/early-exit)
==22249==    by 0x2E7144: rayon_core::current_num_threads (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/early-exit)
==22249==    by 0x290541: <rayon::iter::plumbing::bridge::Callback<C> as rayon::iter::plumbing::ProducerCallback<I>>::callback (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/early-exit)
==22249==    by 0x296AB5: <rayon::iter::while_some::WhileSome<I> as rayon::iter::ParallelIterator>::drive_unindexed (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/early-exit)
==22249==    by 0x29AFB8: rayon::iter::collect::<impl rayon::iter::ParallelExtend<T> for alloc::vec::Vec<T>>::par_extend (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/early-exit)
==22249==    by 0x2B6FF1: rayon::result::<impl rayon::iter::FromParallelIterator<core::result::Result<T,E>> for core::result::Result<C,E>>::from_par_iter (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/early-exit)
==22249== 
==22249== 512 bytes in 1 blocks are still reachable in loss record 21 of 31
==22249==    at 0x4C33E76: memalign (in /usr/lib/valgrind/vgpreload_memcheck-amd64-linux.so)
==22249==    by 0x4C33F91: posix_memalign (in /usr/lib/valgrind/vgpreload_memcheck-amd64-linux.so)
==22249==    by 0x730CD9: aligned_malloc (alloc.rs:95)
==22249==    by 0x730CD9: alloc (alloc.rs:22)
==22249==    by 0x730CD9: __rdl_alloc (alloc.rs:304)
==22249==    by 0x2E9896: alloc::raw_vec::RawVec<T,A>::reserve (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/early-exit)
==22249==    by 0x2E458A: rayon_core::registry::Registry::new (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/early-exit)
==22249==    by 0x2E7D65: std::sync::once::Once::call_once::{{closure}} (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/early-exit)
==22249==    by 0x72B919: std::sync::once::Once::call_inner (once.rs:416)
==22249==    by 0x2E4195: rayon_core::registry::global_registry (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/early-exit)
==22249==    by 0x2E7144: rayon_core::current_num_threads (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/early-exit)
==22249==    by 0x290541: <rayon::iter::plumbing::bridge::Callback<C> as rayon::iter::plumbing::ProducerCallback<I>>::callback (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/early-exit)
==22249==    by 0x296AB5: <rayon::iter::while_some::WhileSome<I> as rayon::iter::ParallelIterator>::drive_unindexed (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/early-exit)
==22249==    by 0x29AFB8: rayon::iter::collect::<impl rayon::iter::ParallelExtend<T> for alloc::vec::Vec<T>>::par_extend (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/early-exit)
==22249== 
==22249== 640 bytes in 1 blocks are still reachable in loss record 22 of 31
==22249==    at 0x4C33E76: memalign (in /usr/lib/valgrind/vgpreload_memcheck-amd64-linux.so)
==22249==    by 0x4C33F91: posix_memalign (in /usr/lib/valgrind/vgpreload_memcheck-amd64-linux.so)
==22249==    by 0x730CD9: aligned_malloc (alloc.rs:95)
==22249==    by 0x730CD9: alloc (alloc.rs:22)
==22249==    by 0x730CD9: __rdl_alloc (alloc.rs:304)
==22249==    by 0x2E47C6: rayon_core::registry::Registry::new (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/early-exit)
==22249==    by 0x2E7D65: std::sync::once::Once::call_once::{{closure}} (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/early-exit)
==22249==    by 0x72B919: std::sync::once::Once::call_inner (once.rs:416)
==22249==    by 0x2E4195: rayon_core::registry::global_registry (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/early-exit)
==22249==    by 0x2E7144: rayon_core::current_num_threads (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/early-exit)
==22249==    by 0x290541: <rayon::iter::plumbing::bridge::Callback<C> as rayon::iter::plumbing::ProducerCallback<I>>::callback (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/early-exit)
==22249==    by 0x296AB5: <rayon::iter::while_some::WhileSome<I> as rayon::iter::ParallelIterator>::drive_unindexed (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/early-exit)
==22249==    by 0x29AFB8: rayon::iter::collect::<impl rayon::iter::ParallelExtend<T> for alloc::vec::Vec<T>>::par_extend (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/early-exit)
==22249==    by 0x2B6FF1: rayon::result::<impl rayon::iter::FromParallelIterator<core::result::Result<T,E>> for core::result::Result<C,E>>::from_par_iter (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/early-exit)
==22249== 
==22249== 640 bytes in 1 blocks are still reachable in loss record 23 of 31
==22249==    at 0x4C33E76: memalign (in /usr/lib/valgrind/vgpreload_memcheck-amd64-linux.so)
==22249==    by 0x4C33F91: posix_memalign (in /usr/lib/valgrind/vgpreload_memcheck-amd64-linux.so)
==22249==    by 0x730CD9: aligned_malloc (alloc.rs:95)
==22249==    by 0x730CD9: alloc (alloc.rs:22)
==22249==    by 0x730CD9: __rdl_alloc (alloc.rs:304)
==22249==    by 0x2F2BF2: std::sync::once::Once::call_once::{{closure}} (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/early-exit)
==22249==    by 0x72B919: std::sync::once::Once::call_inner (once.rs:416)
==22249==    by 0x2F288D: _ZN3std6thread5local4fast12Key$LT$T$GT$14try_initialize17h3d6d30c5335b0a67E.llvm.12191645446024711930 (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/early-exit)
==22249==    by 0x2EA1D2: crossbeam_deque::deque::Stealer<T>::steal (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/early-exit)
==22249==    by 0x2E1C96: <core::iter::adapters::chain::Chain<A,B> as core::iter::traits::iterator::Iterator>::try_fold (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/early-exit)
==22249==    by 0x2E5929: rayon_core::registry::WorkerThread::wait_until_cold (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/early-exit)
==22249==    by 0x2E39FA: rayon_core::registry::ThreadBuilder::run (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/early-exit)
==22249==    by 0x2E6640: std::sys_common::backtrace::__rust_begin_short_backtrace (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/early-exit)
==22249==    by 0x2E0F3E: core::ops::function::FnOnce::call_once{{vtable-shim}} (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/early-exit)
==22249== 
==22249== 808 bytes in 1 blocks are still reachable in loss record 24 of 31
==22249==    at 0x4C31B0F: malloc (in /usr/lib/valgrind/vgpreload_memcheck-amd64-linux.so)
==22249==    by 0x651E0F: alloc::collections::btree::map::BTreeMap<K,V>::insert (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/early-exit)
==22249==    by 0x64A921: wasmer_engine::trap::frame_info::register (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/early-exit)
==22249==    by 0x28A355: <wasmer_engine_jit::artifact::JITArtifact as wasmer_engine::artifact::Artifact>::register_frame_info (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/early-exit)
==22249==    by 0x284625: wasmer_engine::artifact::Artifact::instantiate (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/early-exit)
==22249==    by 0x57FD9E: wasmer::module::Module::instantiate (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/early-exit)
==22249==    by 0x57E713: wasmer::instance::Instance::new (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/early-exit)
==22249==    by 0x263D35: early_exit::main (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/early-exit)
==22249==    by 0x262A66: std::rt::lang_start::{{closure}} (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/early-exit)
==22249==    by 0x732272: {{closure}} (rt.rs:52)
==22249==    by 0x732272: do_call<closure-0,i32> (panicking.rs:348)
==22249==    by 0x732272: try<i32,closure-0> (panicking.rs:325)
==22249==    by 0x732272: catch_unwind<closure-0,i32> (panic.rs:394)
==22249==    by 0x732272: std::rt::lang_start_internal (rt.rs:51)
==22249==    by 0x264881: main (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/early-exit)
==22249== 
==22249== 1,152 bytes in 4 blocks are possibly lost in loss record 25 of 31
==22249==    at 0x4C33B25: calloc (in /usr/lib/valgrind/vgpreload_memcheck-amd64-linux.so)
==22249==    by 0x4013666: allocate_dtv (dl-tls.c:286)
==22249==    by 0x4013666: _dl_allocate_tls (dl-tls.c:530)
==22249==    by 0x5252227: allocate_stack (allocatestack.c:627)
==22249==    by 0x5252227: pthread_create@@GLIBC_2.2.5 (pthread_create.c:644)
==22249==    by 0x7408EC: std::sys::unix::thread::Thread::new (thread.rs:66)
==22249==    by 0x2E0D76: std::thread::Builder::spawn (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/early-exit)
==22249==    by 0x2E4016: <rayon_core::registry::DefaultSpawn as rayon_core::registry::ThreadSpawn>::spawn (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/early-exit)
==22249==    by 0x2E4BDF: rayon_core::registry::Registry::new (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/early-exit)
==22249==    by 0x2E7D65: std::sync::once::Once::call_once::{{closure}} (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/early-exit)
==22249==    by 0x72B919: std::sync::once::Once::call_inner (once.rs:416)
==22249==    by 0x2E4195: rayon_core::registry::global_registry (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/early-exit)
==22249==    by 0x2E7144: rayon_core::current_num_threads (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/early-exit)
==22249==    by 0x290541: <rayon::iter::plumbing::bridge::Callback<C> as rayon::iter::plumbing::ProducerCallback<I>>::callback (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/early-exit)
==22249== 
==22249== 1,520 bytes in 1 blocks are still reachable in loss record 26 of 31
==22249==    at 0x4C31B0F: malloc (in /usr/lib/valgrind/vgpreload_memcheck-amd64-linux.so)
==22249==    by 0x2E4622: rayon_core::registry::Registry::new (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/early-exit)
==22249==    by 0x2E7D65: std::sync::once::Once::call_once::{{closure}} (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/early-exit)
==22249==    by 0x72B919: std::sync::once::Once::call_inner (once.rs:416)
==22249==    by 0x2E4195: rayon_core::registry::global_registry (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/early-exit)
==22249==    by 0x2E7144: rayon_core::current_num_threads (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/early-exit)
==22249==    by 0x290541: <rayon::iter::plumbing::bridge::Callback<C> as rayon::iter::plumbing::ProducerCallback<I>>::callback (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/early-exit)
==22249==    by 0x296AB5: <rayon::iter::while_some::WhileSome<I> as rayon::iter::ParallelIterator>::drive_unindexed (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/early-exit)
==22249==    by 0x29AFB8: rayon::iter::collect::<impl rayon::iter::ParallelExtend<T> for alloc::vec::Vec<T>>::par_extend (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/early-exit)
==22249==    by 0x2B6FF1: rayon::result::<impl rayon::iter::FromParallelIterator<core::result::Result<T,E>> for core::result::Result<C,E>>::from_par_iter (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/early-exit)
==22249==    by 0x2A3A31: <wasmer_compiler_cranelift::compiler::CraneliftCompiler as wasmer_compiler::compiler::Compiler>::compile_module (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/early-exit)
==22249==    by 0x288298: wasmer_engine_jit::artifact::JITArtifact::new (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/early-exit)
==22249== 
==22249== 1,536 bytes in 4 blocks are still reachable in loss record 27 of 31
==22249==    at 0x4C33E76: memalign (in /usr/lib/valgrind/vgpreload_memcheck-amd64-linux.so)
==22249==    by 0x4C33F91: posix_memalign (in /usr/lib/valgrind/vgpreload_memcheck-amd64-linux.so)
==22249==    by 0x730CD9: aligned_malloc (alloc.rs:95)
==22249==    by 0x730CD9: alloc (alloc.rs:22)
==22249==    by 0x730CD9: __rdl_alloc (alloc.rs:304)
==22249==    by 0x2EA0D0: crossbeam_deque::deque::Worker<T>::new_lifo (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/early-exit)
==22249==    by 0x2EB017: core::iter::traits::iterator::Iterator::unzip (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/early-exit)
==22249==    by 0x2E44B6: rayon_core::registry::Registry::new (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/early-exit)
==22249==    by 0x2E7D65: std::sync::once::Once::call_once::{{closure}} (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/early-exit)
==22249==    by 0x72B919: std::sync::once::Once::call_inner (once.rs:416)
==22249==    by 0x2E4195: rayon_core::registry::global_registry (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/early-exit)
==22249==    by 0x2E7144: rayon_core::current_num_threads (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/early-exit)
==22249==    by 0x290541: <rayon::iter::plumbing::bridge::Callback<C> as rayon::iter::plumbing::ProducerCallback<I>>::callback (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/early-exit)
==22249==    by 0x296AB5: <rayon::iter::while_some::WhileSome<I> as rayon::iter::ParallelIterator>::drive_unindexed (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/early-exit)
==22249== 
==22249== 2,008 bytes in 1 blocks are still reachable in loss record 28 of 31
==22249==    at 0x4C31B0F: malloc (in /usr/lib/valgrind/vgpreload_memcheck-amd64-linux.so)
==22249==    by 0x2F233E: crossbeam_epoch::sync::queue::Queue<T>::new (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/early-exit)
==22249==    by 0x2F2BC2: std::sync::once::Once::call_once::{{closure}} (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/early-exit)
==22249==    by 0x72B919: std::sync::once::Once::call_inner (once.rs:416)
==22249==    by 0x2F288D: _ZN3std6thread5local4fast12Key$LT$T$GT$14try_initialize17h3d6d30c5335b0a67E.llvm.12191645446024711930 (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/early-exit)
==22249==    by 0x2EA1D2: crossbeam_deque::deque::Stealer<T>::steal (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/early-exit)
==22249==    by 0x2E1C96: <core::iter::adapters::chain::Chain<A,B> as core::iter::traits::iterator::Iterator>::try_fold (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/early-exit)
==22249==    by 0x2E5929: rayon_core::registry::WorkerThread::wait_until_cold (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/early-exit)
==22249==    by 0x2E39FA: rayon_core::registry::ThreadBuilder::run (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/early-exit)
==22249==    by 0x2E6640: std::sys_common::backtrace::__rust_begin_short_backtrace (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/early-exit)
==22249==    by 0x2E0F3E: core::ops::function::FnOnce::call_once{{vtable-shim}} (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/early-exit)
==22249==    by 0x740BB9: call_once<(),FnOnce<()>> (boxed.rs:1081)
==22249==    by 0x740BB9: call_once<(),alloc::boxed::Box<FnOnce<()>>> (boxed.rs:1081)
==22249==    by 0x740BB9: std::sys::unix::thread::Thread::new::thread_start (thread.rs:87)
==22249== 
==22249== 4,096 bytes in 4 blocks are still reachable in loss record 29 of 31
==22249==    at 0x4C31B0F: malloc (in /usr/lib/valgrind/vgpreload_memcheck-amd64-linux.so)
==22249==    by 0x2EA010: crossbeam_deque::deque::Worker<T>::new_lifo (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/early-exit)
==22249==    by 0x2EB017: core::iter::traits::iterator::Iterator::unzip (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/early-exit)
==22249==    by 0x2E44B6: rayon_core::registry::Registry::new (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/early-exit)
==22249==    by 0x2E7D65: std::sync::once::Once::call_once::{{closure}} (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/early-exit)
==22249==    by 0x72B919: std::sync::once::Once::call_inner (once.rs:416)
==22249==    by 0x2E4195: rayon_core::registry::global_registry (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/early-exit)
==22249==    by 0x2E7144: rayon_core::current_num_threads (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/early-exit)
==22249==    by 0x290541: <rayon::iter::plumbing::bridge::Callback<C> as rayon::iter::plumbing::ProducerCallback<I>>::callback (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/early-exit)
==22249==    by 0x296AB5: <rayon::iter::while_some::WhileSome<I> as rayon::iter::ParallelIterator>::drive_unindexed (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/early-exit)
==22249==    by 0x29AFB8: rayon::iter::collect::<impl rayon::iter::ParallelExtend<T> for alloc::vec::Vec<T>>::par_extend (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/early-exit)
==22249==    by 0x2B6FF1: rayon::result::<impl rayon::iter::FromParallelIterator<core::result::Result<T,E>> for core::result::Result<C,E>>::from_par_iter (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/early-exit)
==22249== 
==22249== 6,080 bytes in 4 blocks are still reachable in loss record 30 of 31
==22249==    at 0x4C31B0F: malloc (in /usr/lib/valgrind/vgpreload_memcheck-amd64-linux.so)
==22249==    by 0x2E33C8: rayon_core::registry::ThreadBuilder::run (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/early-exit)
==22249==    by 0x2E6640: std::sys_common::backtrace::__rust_begin_short_backtrace (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/early-exit)
==22249==    by 0x2E0F3E: core::ops::function::FnOnce::call_once{{vtable-shim}} (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/early-exit)
==22249==    by 0x740BB9: call_once<(),FnOnce<()>> (boxed.rs:1081)
==22249==    by 0x740BB9: call_once<(),alloc::boxed::Box<FnOnce<()>>> (boxed.rs:1081)
==22249==    by 0x740BB9: std::sys::unix::thread::Thread::new::thread_start (thread.rs:87)
==22249==    by 0x52516DA: start_thread (pthread_create.c:463)
==22249==    by 0x57A271E: clone (clone.S:95)
==22249== 
==22249== 8,160 bytes in 4 blocks are still reachable in loss record 31 of 31
==22249==    at 0x4C31B0F: malloc (in /usr/lib/valgrind/vgpreload_memcheck-amd64-linux.so)
==22249==    by 0x2EFC1F: crossbeam_epoch::internal::Local::register (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/early-exit)
==22249==    by 0x2F2814: _ZN3std6thread5local4fast12Key$LT$T$GT$14try_initialize17h3d6d30c5335b0a67E.llvm.12191645446024711930 (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/early-exit)
==22249==    by 0x2EA1D2: crossbeam_deque::deque::Stealer<T>::steal (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/early-exit)
==22249==    by 0x2E1C96: <core::iter::adapters::chain::Chain<A,B> as core::iter::traits::iterator::Iterator>::try_fold (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/early-exit)
==22249==    by 0x2E5929: rayon_core::registry::WorkerThread::wait_until_cold (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/early-exit)
==22249==    by 0x2E39FA: rayon_core::registry::ThreadBuilder::run (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/early-exit)
==22249==    by 0x2E6640: std::sys_common::backtrace::__rust_begin_short_backtrace (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/early-exit)
==22249==    by 0x2E0F3E: core::ops::function::FnOnce::call_once{{vtable-shim}} (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/early-exit)
==22249==    by 0x740BB9: call_once<(),FnOnce<()>> (boxed.rs:1081)
==22249==    by 0x740BB9: call_once<(),alloc::boxed::Box<FnOnce<()>>> (boxed.rs:1081)
==22249==    by 0x740BB9: std::sys::unix::thread::Thread::new::thread_start (thread.rs:87)
==22249==    by 0x52516DA: start_thread (pthread_create.c:463)
==22249==    by 0x57A271E: clone (clone.S:95)
==22249== 
==22249== LEAK SUMMARY:
==22249==    definitely lost: 0 bytes in 0 blocks
==22249==    indirectly lost: 0 bytes in 0 blocks
==22249==      possibly lost: 1,152 bytes in 4 blocks
==22249==    still reachable: 29,232 bytes in 87 blocks
==22249==         suppressed: 0 bytes in 0 blocks
==22249== 
==22249== For counts of detected and suppressed errors, rerun with: -v
==22249== ERROR SUMMARY: 2 errors from 2 contexts (suppressed: 0 from 0)
